src/main/java/emt/item/focus/ItemMaintenanceFocus.java

```diff
@@ -0,0 +1,72 @@
+package emt.item.focus;
+
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.MovingObjectPosition;
+import net.minecraft.world.World;
+
+import thaumcraft.api.aspects.Aspect;
+import thaumcraft.api.aspects.AspectList;
+import thaumcraft.api.wands.FocusUpgradeType;
+import thaumcraft.common.items.wands.ItemWandCasting;
+import gregtech.api.metatileentity.BaseMetaTileEntity;
+import gregtech.api.metatileentity.implementations.GT_MetaTileEntity_Hatch_Maintenance;
+
+public class ItemMaintenanceFocus extends ItemBaseFocus {
+
+    private static final AspectList visCost = new AspectList().add(Aspect.FIRE, 2500).add(Aspect.EARTH, 2500)
+            .add(Aspect.ORDER, 2500);
+
+    public ItemMaintenanceFocus() {
+        super("maintenance");
+    }
+
+    @Override
+    public int getFocusColor(ItemStack stack) {
+        return 0xC0C0C0;
+    }
+
+    @Override
+    public AspectList getVisCost(ItemStack focusstack) {
+        return visCost;
+    }
+
+    @Override
+    public String getSortingHelper(ItemStack itemstack) {
+        return "MAINTENANCE";
+    }
+
+    @Override
+    public FocusUpgradeType[] getPossibleUpgradesByRank(ItemStack focusstack, int rank) {
+        return new FocusUpgradeType[] { FocusUpgradeType.frugal };
+    }
+
+    @Override
+    public ItemStack onFocusRightClick(ItemStack itemStack, World world, EntityPlayer player,
+            MovingObjectPosition mop) {
+
+        ItemWandCasting wand = (ItemWandCasting) itemStack.getItem();
+
+        if (!world.isRemote) {
+            if (player.isSneaking()) {
+                if (((BaseMetaTileEntity) world.getTileEntity(mop.blockX, mop.blockY, mop.blockZ))
```

Comment:
Check if TE is an instance of `BaseMetaTileEntity`. Currently it crashes when you use wand to grass block

Reply:
Hmm I thought I check this. Let me double check that it's crashing when that happens.

Reply:
Ha, you're right. I somehow missed that. I simplified the code earlier and fucked that up it seems.